homeassistant/components/blink/__init__.py

```diff
@@ -28,17 +28,36 @@
 SERVICE_SEND_PIN_SCHEMA = vol.Schema({vol.Optional(CONF_PIN): cv.string})
 
 
-def _blink_startup_wrapper(entry):
+def _blink_startup_wrapper(hass, entry):
     """Startup wrapper for blink."""
     blink = Blink()
-    blink.auth = Auth(deepcopy(dict(entry.data)), no_prompt=True)
+    auth_data = deepcopy(dict(entry.data))
+    blink.auth = Auth(auth_data, no_prompt=True)
     blink.refresh_rate = entry.options.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
 
     if blink.start():
         blink.setup_post_verify()
+    elif blink.auth.check_key_required():
+        _LOGGER.debug("Attempting a reauth flow.")
+        _reauth_flow_wrapper(hass, auth_data)
+
     return blink
 
 
+def _reauth_flow_wrapper(hass, data):
+    """Reauth flow wrapper."""
+    hass.async_create_task(
```

Comment:
We're in sync context here so can't use the async api. Either use `hass.add_job` instead or move this to an async context.